cirq/sim/clifford/clifford_simulator.py

```diff
@@ -331,21 +333,34 @@ def wave_function(self):
         return self.state_vector()
 
     def apply_unitary(self, op: 'cirq.Operation'):
-        tableau_args = clifford.ActOnCliffordTableauArgs(
-            self.tableau, [self.qubit_map[i] for i in op.qubits], np.random.RandomState(), {}
-        )
         ch_form_args = clifford.ActOnStabilizerCHFormArgs(
             self.ch_form, [self.qubit_map[i] for i in op.qubits], np.random.RandomState(), {}
         )
         try:
-            act_on(op, tableau_args)
             act_on(op, ch_form_args)
         except TypeError:
             raise ValueError(
                 '%s cannot be run with Clifford simulator.' % str(op.gate)
             )  # type: ignore
         return
 
+    def apply_measurement(
+        self,
+        op: 'cirq.Operation',
+        measurements: Dict[str, List[np.ndarray]],
+        prng: np.random.RandomState,
+        collapse_state_vector=True,
+    ):
+        if collapse_state_vector:
```

Comment:
as this is a public method, I think we should ensure that the op passed in has a MeasurementGate as op.gate. WDYT?

Reply:
Makes sense. Added the validation and some tests.